### ISSUE TITLE
Discrepancy in epoch between `fourier_fit_magseries` and `fourier_sinusoidal_func`

### ISSUE THREAD

**Issue by joshuawallace** [waqasbhatti/astrobase#70]
`fourier_fit_magseries()` returns an epoch that corresponds to the minimum brightness in the light curve ([here]( https://github.com/waqasbhatti/astrobase/blob/master/astrobase/varbase/lcfit.py#L435)).  'fourier_sinusoidal_func()' makes no such assumption for the epoch ([here](https://github.com/waqasbhatti/astrobase/blob/master/astrobase/lcmodels/sinusoidal.py#L50), and line 60 where `phase` gets used).  In my application of the code, where I'm fitting a Fourier series to an RR Lyrae signal and then inserting an attenuated version of that signal into some other light curves, this leads to the inserted signal not having the same epoch as the original signal.  What should I do to fix this?

**Comment by waqasbhatti** [comment]
Sorry for the delay, was away for a couple of days.

I think the easiest way to do this would be to pass in the epoch you get from the `fourier_fit_magseries` function to the `fourier_sinusoidal_func` function as part of the input `fourierparams`. That should make it use the same epoch as the one you got out of the fit. If you want to use another epoch (e.g. that of the object whose LC you're inserting the signal into), I think passing that in as part of the `fourierparams` will do the right thing.

**Comment by joshuawallace** [comment]
That's what I'm doing currently and having a problem with.  I'm using the [`['fitinfo']['fitepoch']`](https://github.com/waqasbhatti/astrobase/blob/master/astrobase/varbase/lcfit.py#L446) from `fourier_fit_magseries`'s returndict as the epoch for `fourier_sinusoidal_func` and it leads to the creation of a signal that does not line up with the signal in the light curve being fit by 'fourier_fit_magseries'.

**Comment by waqasbhatti** [comment]
Ah, I see what you mean now. A random RRab from HATSouth:

![magseries-phased-plot](https://user-images.githubusercontent.com/1868184/49041311-db34f400-f192-11e8-88ea-32d896a3b092.png)

The model fit:

![magseries-phased-plot](https://user-images.githubusercontent.com/1868184/49041323-ebe56a00-f192-11e8-93e3-8fe885673244.png)

Let me think about this for a bit. I thought I had this working earlier...

**Comment by joshuawallace** [comment]
I think I figured it out.  The `_get_phased_quantities` function in `fourier_fit_magseries` uses the *minimum time* as the epoch for phase folding.  This value does not get returned out of `fourier_fit_magseries`, and the value that does get returned (time of minimum in light curve) does not correspond to the actual epoch for the phase-folding and fitting, and neither does t=0, which I tried.

I will change the code so that the returned `['fitinfo']['fitepoch']` value is the same as used by `_get_phased_quantities` and if it works out, I'll create a pull request.
